src/core/eme/index.ts

```diff
@@ -183,13 +170,17 @@ function createEME(
         throw new EncryptedMediaError("INVALID_ENCRYPTED_EVENT", error, true);
       }
 
-      const sessionEvents = createOrReuseSessionWithRetry(
-        encryptedEvent,
+      const initData = new Uint8Array(encryptedEvent.initData);
+      const initDataType = encryptedEvent.initDataType;
+
+      const events$ = createOrReuseSessionWithRetry(
```

Comment:
Here we create the session before setting the MediaKeys, though we did both of them at the same time before.

I'm telling this because we had issues with ChromeCast before, where a deadlock was created in that kind of scenario (even if the spec says nothing of that).
Also, it can be an optimization to do that.

Moreover, I do not get why we wait for the first ``createOrReuseSessionWithRetry``' event to attach the MediaKeys to the video element.

And I get even less why we re-feed the session to ``handleSessionEvents`` each time ``createOrReuseSessionWithRetry`` emits.
Both ``handleSessionEvents`` and ``createOrReuseSessionWithRetry`` return the same events (``Observable<ISessionEvent>``) so this is confusing.